services/snapshotter/service.go

```diff
@@ -133,9 +133,20 @@ func (s *Service) handleConn(conn net.Conn) error {
 	return nil
 }
 
-func (s *Service) writeMetaStore(conn net.Conn) error {
+func (s *Service) writeMetaStore(conn net.Conn, database string) error {
 	// Retrieve and serialize the current meta data.
-	metaBlob, err := s.MetaClient.MarshalBinary()
+	// if the database name is non-empty, then we drop all the DB's from the metadata
+	// that aren't being exported.
+	data := s.MetaClient.Data()
+	if database != "" {
+		for _, db := range data.Databases[:] {
```

Comment:
ok, removed for other reasons but what I was trying to avoid (and I see my error now) was modifying the array that I was iterating.  So I needed a copy.   I got a bit pythonic there, need to unlearn some things :)